src/components/inputs/InputContainer.tsx

```diff
@@ -13,10 +13,7 @@ interface InputContainerProps {
 
 const InputContainer = memo(
     ({ children, hasHandle, id, index, label }: React.PropsWithChildren<InputContainerProps>) => {
-        const isValidConnection = useContextSelector(
-            GlobalVolatileContext,
-            (c) => c.isValidConnection
-        );
+        const { isValidConnection } = useContext(GlobalContext);
```

Comment:
How come you didn't need the context selector anymore?

Reply:
I'm assuming it's because it's memoized better now?

Reply:
Yes, isValidConnection only depends on `schemata, getNode, getNodes, getEdges` now. All of these should be pretty constant.